sc-memory/sc-memory/sc_keynodes.cpp

```diff
@@ -48,22 +48,82 @@ ScAddr ScKeynodes::kBinaryUInt64;
 ScAddr ScKeynodes::kBinaryString;
 ScAddr ScKeynodes::kBinaryCustom;
 
-bool ScKeynodes::Init(bool force)
+bool ScKeynodes::Init(bool force, sc_char const * result_structure_system_idtf)
 {
   if (ms_isInitialized && !force)
     return true;
 
   bool result = ScKeynodes::InitGlobal();
   ScMemoryContext ctx(sc_access_lvl_make_min, "ScKeynodes::Init");
 
+  ScAddrVector const & keynodesAddrs = {
+      kCommandStateAddr,
+      kCommandInitiatedAddr,
+      kCommandProgressedAddr,
+      kCommandFinishedAddr,
+      kNrelResult,
+      kNrelCommonTemplate,
+      kNrelIdtf,
+      kNrelFormat,
+      kScResult,
+      kBinaryType};
+
+  ScAddrVector const & resultCodes = {
+      kScResultError,
+      kScResultOk,
+      kScResultErrorInvalidParams,
+      kScResultErrorInvalidType,
+      kScResultErrorIO,
+      kScResultInvalidState,
+      kScResultErrorNotFound,
+      kScResultErrorNoWriteRights,
+      kScResultErrorNoReadRights,
+      kScResultNo,
+      kScResultUnknown};
+
+  ScAddrVector const & binaryTypes = {
+      kBinaryDouble,
+      kBinaryFloat,
+      kBinaryString,
+      kBinaryInt8,
+      kBinaryInt16,
+      kBinaryInt32,
+      kBinaryInt64,
+      kBinaryUInt8,
+      kBinaryUInt16,
+      kBinaryUInt32,
+      kBinaryUInt64,
+      kBinaryCustom};
+
+  std::vector<ScAddrVector> const & keynodeVectors = {keynodesAddrs, resultCodes, binaryTypes};
+
+  ScAddr resultStruct;
+  bool resultStructValid = SC_FALSE;
+
+  if (result_structure_system_idtf != (sc_char const *)null_ptr)
+  {
+    resultStruct = ctx.HelperResolveSystemIdtf(result_structure_system_idtf, ScType::NodeConstStruct);
+    resultStructValid = resultStruct.IsValid();
+
+    for (ScAddrVector keynodes : keynodeVectors)
+    {
+      for (ScAddr keynodeAddr : keynodes)
+      {
+        ctx.CreateEdge(ScType::EdgeAccessConstPosPerm, resultStruct, keynodeAddr);
+      }
+    }
+  }
+
   // init sc_result set
-  for (size_t i = 0; i < SC_RESULT_COUNT; ++i)
+  for (ScAddr const & resAddr : resultCodes)
   {
-    ScAddr const resAddr = GetResultCodeAddr(static_cast<sc_result>(i));
     result = result && resAddr.IsValid();
```

Comment:
```suggestion
    result &= resAddr.IsValid();
```
else you create tempory object

Reply:
Check it everywhere and correct